tests/h/validators_test.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+import colander
+from mock import Mock
+import pytest
+
+from h.validators import Email
+
+
+class TestEmail(object):
+    @pytest.mark.parametrize('email', [
+        'jimsmith@foobar.com',
+        'international@xn--domain.com',
+        'jim.smith@gmail.com',
+        'jim.smith@foo.bar.com',
+        'a.range!of#punctuation-chars$are+accepted@gmail.com',
+    ])
+    def test_accepts_valid_addresses(self, schema_node, email):
+        validator = Email()
+        validator(schema_node, email)
+
+    @pytest.mark.parametrize('email', [
+        ' spaces @ spaces.com',
+        'sorry-ascii-only-🤠@gmail.com',
```

Comment:
Hmm, these are not unicode strings at the moment. Maybe they should be? By either adding the `unicode_literals` import, or specifically using `u''`.

Reply:
Good point. I've added the `unicode_literals` import since we use that widely elsewhere in the "h" code.